tests/providers/databricks/sensors/test_databricks_partition.py

```diff
@@ -99,3 +99,67 @@ def test_unsupported_conn_type(self):
     def test_partition_sensor(self, patched_poke):
         patched_poke.return_value = True
         assert self.partition_sensor.poke({})
+
+    @pytest.mark.parametrize(
+        "soft_fail, expected_exception", ((False, AirflowException), (True, AirflowSkipException))
+    )
+    def test_fail__generate_partition_query(self, soft_fail, expected_exception):
+        self.partition_sensor.soft_fail = soft_fail
+        table_name = "test"
+        with pytest.raises(expected_exception) as e, patch(
+            "airflow.providers.databricks.sensors.databricks_partition.DatabricksPartitionSensor"
+            "._sql_sensor"
+        ) as _sql_sensor:
+            _sql_sensor.return_value = [[[], [], [], [], [], [], [], []]]
+            self.partition_sensor._generate_partition_query(
+                prefix="", suffix="", joiner_val="", table_name=table_name
+            )
+        assert e.match(f"Table {table_name} does not have partitions")
```

Comment:
should we just move `match` to `pytest.raises(expected_exception, match=...)`?

Reply:
Made the required change.